### PR TITLE
test: migrated reply-early-hints.test.js from tap to node:test

### PR DESCRIPTION
**Pull request by Tony133**
#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

Proposal:

  - rename file in `kebab-case`
  - migrated `reply-early-hints.test.js` file from `tap` to `node:test` 🔥

### CODE REVIEW

File: test/reply-early-hints.test.js [hunk 1]
```diff
@@ -88,7 +84,7 @@ test('sends early hints (http2)', (t) => {
     })
 
     req.on('end', () => {
-      t.equal(data, testResBody)
+      t.assert.strictEqual(data, testResBody)
       client.close()
       fastify.close(t.end)
```
Comment (dancastillo) [inline test/reply-early-hints.test.js:89]: `t.end` is a `tap` method we should remove this
```suggestion
      fastify.close()
```

File: test/reply-early-hints.test.js [hunk 2]
```diff
@@ -18,41 +16,39 @@ test('sends early hints', (t) => {
...
-        t.equal(data, testResBody)
+        t.assert.strictEqual(data, testResBody)
         fastify.close(t.end)
       })
     })
   })
 })
 
 test('sends early hints (http2)', (t) => {
-  t.plan(6)
```
Comment (dancastillo) [inline test/reply-early-hints.test.js:54]: can we add this back in

Comment (Tony133) [reply test/reply-early-hints.test.js:54]: I removed it because it was failing the test, but if you want I will try to put it back

Comment (dancastillo) [reply test/reply-early-hints.test.js:54]: if you pass done in so it is `test('......', (t, done) => {})` and call it at end should follow the plan assertion count

like this
```js
'use strict'

const Fastify = require('..')
const { test } = require('node:test')
const http = require('http')
const http2 = require('http2')

const testResBody = 'Hello, world!'

test('sends early hints', (t, done) => {
  t.plan(6)

  const fastify = Fastify({
    logger: false
  })

  fastify.get('/', async (request, reply) => {
    reply.writeEarlyHints({
      link: '</styles.css>; rel=preload; as=style'
    }, () => {
      t.assert.ok('callback called')
    })

    return testResBody
  })

  fastify.listen({ port: 0 }, (err, address) => {
    t.assert.ifError(err)

    const req = http.get(address)

    req.on('information', (res) => {
      t.assert.strictEqual(res.statusCode, 103)
      t.assert.strictEqual(res.headers.link, '</styles.css>; rel=preload; as=style')
    })

    req.on('response', (res) => {
      t.assert.strictEqual(res.statusCode, 200)

      let data = ''
      res.on('data', (chunk) => {
        data += chunk
      })

      res.on('end', () => {
        t.assert.strictEqual(data, testResBody)
        fastify.close()
        done()
      })
    })
  })
})

test('sends early hints (http2)', (t, done) => {
  t.plan(6)

  const fastify = Fastify({
    http2: true,
    logger: false
  })

  fastify.get('/', async (request, reply) => {
    reply.writeEarlyHints({
      link: '</styles.css>; rel=preload; as=style'
    })

    return testResBody
  })

  fastify.listen({ port: 0 }, (err, address) => {
    t.assert.ifError(err)

    const client = http2.connect(address)
    const req = client.request()

    req.on('headers', (headers) => {
      t.assert.notStrictEqual(headers, undefined)
      t.assert.strictEqual(headers[':status'], 103)
      t.assert.strictEqual(headers.link, '</styles.css>; rel=preload; as=style')
    })

    req.on('response', (headers) => {
      t.assert.strictEqual(headers[':status'], 200)
    })

    let data = ''
    req.on('data', (chunk) => {
      data += chunk
    })

    req.on('end', () => {
      t.assert.strictEqual(data, testResBody)
      client.close()
      fastify.close()
      done()
    })

    req.end()
  })
})
```

Comment (Tony133) [reply test/reply-early-hints.test.js:54]: I updated the test, with the changes you told me, the checks all passed, thanks! 😉

**Pull request merged**